apps/lukso-status/src/app/services/vanguard-metrics.service.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, of, timer } from 'rxjs';
+import { catchError, map, switchMap, tap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class VanguardService {
+  metrics$: Observable<any>;
+  constructor(private httpClient: HttpClient) {
+    this.metrics$ = timer(0, 3000).pipe(
+      switchMap(() => {
+        return httpClient
+          .get('/vanguard/metrics', {
+            responseType: 'text',
+          })
+          .pipe(
```

Comment:
have you ever heard of `async/await`? : )

Reply:
That's not applicable here, Angular uses RxJS which is way more powerful. 👍